src/printer.ts

```diff
@@ -194,6 +199,48 @@ export class PugPrinter {
 						results.push(this[token.type](token));
 						break;
 				}
+				if (
+					token.type === 'comment' &&
```

Comment:
Can all this logic maybe moved to `comment` member-function down in this file? Or is there a reason why this should be done here? Maybe cause you accessing the `index`? :thinking:  

https://github.com/prettier/plugin-pug/blob/48dfff17416f21bc3fe661d2e4a8d9265604e2f9/src/printer.ts#L736

Reply:
I was trying this approach first, but yes the problem is that it has to increase the index and skip all the processing of all the other tokens, so it seemed easer to do it outside of the normal processing loop.

Reply:
I found a way to make this work! Check the last two commits, I think it's very clean now.